Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp

```diff
@@ -1146,435 +1272,672 @@ void from_hex(const std::string &s, unsigned char *data, int len) {
 
-    } catch (const rust::cxxbridge1::Error &e) {
-        success = false;
-        output_message =
-            FString::Printf(TEXT("CronosPlayUnreal Erc20Balance Error: %s"),
-                            UTF8_TO_TCHAR(e.what()));
-    }
+        try {
+            std::string mycontractaddress = TCHAR_TO_UTF8(*contractAddress);
+            std::string myaccountaddress = TCHAR_TO_UTF8(*accountAddress);
+            std::string mycronosrpc = TCHAR_TO_UTF8(*myCronosRpc);
+
+            Erc20 erc20 =
+                new_erc20(mycontractaddress, mycronosrpc, myCronosChainID)
+                    .legacy();
+            U256 erc20_balance = erc20.balance_of(myaccountaddress);
+            balance = UTF8_TO_TCHAR(erc20_balance.to_string().c_str());
+            success = true;
+        } catch (const rust::cxxbridge1::Error &e) {
+            output_message =
+                FString::Printf(TEXT("CronosPlayUnreal Erc20Balance Error: %s"),
+                                UTF8_TO_TCHAR(e.what()));
+        }
+
+        AsyncTask(ENamedThreads::GameThread,
+                  [Out, balance, success, output_message]() {
+                      Out.ExecuteIfBound(balance, success, output_message);
+                  });
+    });
 }
 
 void ADefiWalletCoreActor::Erc721Balance(FString contractAddress,
                                          FString accountAddress,
-                                         FString &balance, bool &success,
-                                         FString &output_message) {
-    try {
-        std::string mycontractaddress = TCHAR_TO_UTF8(*contractAddress);
-        std::string myaccountaddress = TCHAR_TO_UTF8(*accountAddress);
-        std::string mycronosrpc = TCHAR_TO_UTF8(*myCronosRpc);
-        Erc721 erc721 =
-            new_erc721(mycontractaddress, mycronosrpc, myCronosChainID)
-                .legacy();
-        U256 erc721_balance = erc721.balance_of(myaccountaddress);
-        balance = UTF8_TO_TCHAR(erc721_balance.to_string().c_str());
-        success = true;
+                                         FGeneralCronosCallDelegate Out) {
+    AsyncTask(ENamedThreads::AnyHiPriThreadNormalTask, [this, contractAddress,
+                                                        accountAddress, Out]() {
+        FString balance;
+        FString output_message;
+        bool success = false;
 
-    } catch (const rust::cxxbridge1::Error &e) {
-        success = false;
-        output_message =
-            FString::Printf(TEXT("CronosPlayUnreal Erc721Balance Error: %s"),
-                            UTF8_TO_TCHAR(e.what()));
-    }
+        try {
+            std::string mycontractaddress = TCHAR_TO_UTF8(*contractAddress);
+            std::string myaccountaddress = TCHAR_TO_UTF8(*accountAddress);
+            std::string mycronosrpc = TCHAR_TO_UTF8(*myCronosRpc);
+            Erc721 erc721 =
+                new_erc721(mycontractaddress, mycronosrpc, myCronosChainID)
+                    .legacy();
```

Comment:
why does `.legacy()` need to be called?

Reply:
oh I see it in many places, but it was in the original code

Reply:
legacy: not use  EIP-1559
not legacy: use  EIP-1559

default is using  EIP-1559

i'll check with  EIP-1559 off, and remove legacy()

Reply:
1) legacy isn't needed for Cronos, 2) this flag is turned even on query calls, where no transaction is submitted, so it doesn't make a difference there

Reply:
yes, i'll remove .legacy()